### PR TITLE
docs(user): add the Tutorials chapter

### PR DESCRIPTION
**Pull request by lyfne123**
## What

Adds `docs/{en,zh}/user/tutorials/` — seven task-oriented walkthroughs (B3), plus the three examples they needed.

**Writing operators**

| Page | Ends in |
| ---- | ------- |
| `00-elementwise.md` | An element-wise kernel, tensor level → tile level → chunked → checked |
| `01-reduction-softmax.md` | A numerically stable softmax |
| `02-matmul.md` | A K-blocked matmul, and what split-K costs |
| `03-mixed-kernel.md` | Cube and vector overlapping in one scope |

**Shaping the schedule**

| Page | Ends in |
| ---- | ------- |
| `04-task-graph.md` | A program whose dependency graph you control |
| `05-scheduling-tuning.md` | A measurement loop over the four DFX observation points |

## The mixed-kernel gap

`03-mixed-kernel.md` closes the gap #2120 flagged as larger than distributed. There was no standalone mixed-kernel example anywhere — the only real one is inside `examples/models/qwen3_jit/kernels/projection.py`, and `examples/intermediate/01_fused_linear.py`-style "fused" kernels are two sequential InCore functions, which is the form a mixed kernel replaces.

The page documents the cross-core ring sizing that every non-toy mixed kernel runs into:

| | |
| --- | --- |
| Tile crossing the boundary | `[128, 128]` FP32 = 64 KB |
| Default ring depth | 8 slots → **512 KB** |
| Vector budget | **184 KB** |

So `pl.cross_core_slot(slot_num=2)` is mandatory rather than decorative — and the error, which names a byte count and not the tile, is quoted verbatim so a reader can match it.

## New examples

- `examples/advanced/03_mixed_kernel.py` — three modes (staged / UP_DOWN / LEFT_RIGHT)
- `examples/intermediate/07_task_graph.py` — an inferred edge and the same edge declared
- `examples/beginner/02_elementwise.py` — extended with `chunked_add`, the step from one tile to a tensor that needs a loop

## Verification

Every runnable example was **executed on a2a3sim**, not just compiled:

```
19 runnable example(s), 0 failed
```

Examples carrying their own `torch.allclose` have that assertion run as part of the check, so the softmax, matmul, mixed-kernel and task-graph pages are golden-checked, not merely non-NaN.

Gates: `check_docs_symbol_coverage` / `check_docs_en_zh_parity` (136 paired) / `check_docs_nav` (136 entries) / `check_english_only` all exit 0; markdownlint 0 errors; no link problems introduced.

`docs/{en,zh}/user/ops/01-catalog.md` said the mixed-kernel tutorial "is not written yet" — that now links here.

## Note

This branch predates #2347. Its examples already use `pl.assemble` throughout, so there is no overlap in content, only in subject.

### CODE REVIEW

File: docs/en/user/tutorials/03-mixed-kernel.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+b = torch.randn(256, 128, dtype=torch.float16)
+bias = torch.randn(128, 128, dtype=torch.float32)
+out = torch.zeros(128, 128, dtype=torch.float32)
+mixed(a, b, bias, out, config=RunConfig(platform="a2a3sim"))
+assert torch.allclose(out, a.float() @ b.float() + bias, rtol=1e-2, atol=1e-2)
+```
+
+`pl.split(mode)` marks the scope as mixed. The compiler halves the work along the chosen
+axis, gives the cube one half and the vector units the other, and inserts the cross-core
+transfers that move results between them.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/03-mixed-kernel.md]: Fixed in 96f3c249. Confirmed against `docs/en/dev/passes/20-lower_auto_vector_split.md:6` ("halves only the **vector sub-region**"). The page described the cube and vector units as each taking half of one tile, which is the wrong execution model; it now says the cube sub-region stays full-sized and the split shards vector work across the two AIV lanes.

File: docs/en/user/tutorials/index.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+| `tpush_to_aiv`, `tpop_from_aic`, `aiv_shard`, `aic_gather` | Both, by construction | [Mixed kernels](03-mixed-kernel.md) |
+
+A kernel built only from one family occupies one unit and leaves the other idle. That
+observation is the whole point of [Mixed kernels](03-mixed-kernel.md); everything before it
+writes single-unit kernels. See [Operations](../ops/01-catalog.md) for the full list.
+
+## Running the examples
+
+Every page names a file under `examples/`. They take a `--platform` flag and default to the
+simulator, so nothing here needs a device:
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/index.md]: Fixed in 96f3c249. Checked each companion: only `03_mixed_kernel.py` has argparse. The page now says `RunConfig.platform` defaults to `"a2a3sim"` — which is what actually keeps them off a device — and names `03_mixed_kernel.py` as the one exception that takes flags.

File: docs/zh/user/tutorials/00-elementwise.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+这里有三件事在起作用。
+
+**`pl.Out[...]` 是方向，不是提示。** 它告诉编译器这个参数被写、不被读。运行时正是据此把本任务与其他任务定序 —— 见 [依赖模型](../tasks/00-model.md)。写错了得到的是竞态，不是报错。
+
+**`with pl.at(level=pl.Level.CORE_GROUP):` 才是放算子的地方。** `@pl.jit` 函数体本身是控制面；它派发工作，但不能容纳算子。把 `out = pl.add(a, b)` 写在块外是错误 —— 解析器会说。
+
+**`pl.assemble` 才是写输出的方式。** 这一步是所有人都会栽的地方：
+
+```python
+out = pl.add(a, b)                              # ✗ 编译通过，什么都没写
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/zh/user/tutorials/00-elementwise.md]: Fixed in 96f3c249. The comments inside the zh code blocks are English again, per `.claude/rules/documentation.md` ("Code examples are NOT translated"). Added a check over all zh tutorial code blocks to confirm none remain.

File: docs/en/user/tutorials/00-elementwise.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+**`pl.assemble` is how you write an output.** This is the step everyone gets wrong:
+
+```python
+out = pl.add(a, b)                              # ✗ compiles, writes nothing
+out = pl.assemble(out, pl.add(a, b), [0, 0])    # ✓
+```
+
+The first line rebinds a local name. It compiles, it runs, and the output buffer keeps
+whatever was in it — you get NaN, with no diagnostic from any stage. `pl.assemble(dst, value,
+offset)` is the operator that actually places `value` into `dst` at `offset`.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/00-elementwise.md]: Reworded in 96f3c249, though not for the stated reason. I measured this on a2a3sim with a zero-initialised output and the result was NaN, not stale zeros — the caller's buffer is not what comes back. Your underlying point stands: I should not promise NaN. The text now says the buffer is never written and you get whatever it held, noting NaN is what the simulator returned here.

File: docs/en/user/tutorials/01-reduction-softmax.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+`row_expand_expdif` is the fused form of steps 2 and 3 below. Reach for it once the
+unfused version works.
+
+## Step 3: the whole softmax
+
+Softmax is `exp(x) / sum(exp(x))`, but computed that way it overflows: `exp(88)` already
+leaves FP32 range. Subtracting the row maximum first is mathematically a no-op — the
+`exp(-max)` factors cancel between numerator and denominator — and keeps every exponent
+at or below zero.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/01-reduction-softmax.md]: Fixed in 96f3c249. Correct — `exp(88)` is ~1.65e38, inside FP32's ~3.4e38 range. Now states the ceiling and uses `exp(89)`.

File: docs/en/user/tutorials/01-reduction-softmax.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+torch.manual_seed(0)
+a = torch.randn(64, 64)
+out = torch.zeros(64, 64)
+softmax(a, out, config=RunConfig(platform="a2a3sim"))
+assert torch.allclose(out, torch.softmax(a, dim=1), rtol=1e-5, atol=1e-5)
+```
+
+Two scratch tiles, because the two reductions are live at different times and each needs
+its own working space.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/01-reduction-softmax.md]: Fixed in 96f3c249. The justification was backwards: non-overlapping lifetimes are exactly when one scratch *can* serve both. The text now says so and explains the example keeps them separate for readability.

File: docs/en/user/tutorials/02-matmul.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+```python
+with pl.at(level=pl.Level.CORE_GROUP, name_hint="zero_init"):
+    c = pl.assemble(c, pl.full([M, N], dtype=pl.FP32, value=0.0), [0, 0])
+for ks in pl.parallel(SPLITS):
+    with pl.at(level=pl.Level.CORE_GROUP, name_hint="split_k"):
+        partial = pl.matmul(a_k, b_k, out_dtype=pl.FP32)
+        c = pl.assemble(c, partial, [0, 0], atomic=pl.AtomicType.Add)
+```
+
+Fragment — the runnable version is `examples/advanced/01_split_k.py`.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/02-matmul.md]: Fixed in 518d38c1. The fragment now derives `k0 = ks * KS` and slices `a`/`b` by it, so the loop variable does the work the prose claims. Also names which identifiers come from the enclosing function, and adds that every core writes the whole `[M, N]` output — it is K that is divided.

File: docs/en/user/tutorials/02-matmul.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+| Aspect | Cost |
+| ------ | ---- |
+| Zero-init | The output must be zeroed first; atomic add has no "first writer" |
+| Determinism | **Accumulation order across cores is not fixed.** Repeated runs give bitwise different results |
+| When it pays | K is large and M/N are too small to fill the cores on their own |
+
+That second row is the one to weigh. If a downstream test compares bitwise, or you are
+chasing a numerical discrepancy, split-K makes the answer a moving target. Use it when the
+parallelism is worth more than reproducibility.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/02-matmul.md:121]: Fixed in 96f3c249. Changed to "may differ in the last bits" in both languages.

File: docs/en/user/tutorials/04-task-graph.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+The runtime does not execute your orchestration function statement by statement. It builds
+a dependency graph and runs whatever is ready.
+
+> **Statement order expresses nothing.** Two dispatches written one after the other are
+> ordered only if *something* says so — a buffer overlap the runtime can see, or an edge
+> you declared.
+
+Where the edges come from by default: the runtime records which buffers each task touches
+and how, using each parameter's direction, and derives an edge wherever two tasks touch the
+same buffer.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/04-task-graph.md:24]: Fixed in 518d38c1. You are right that the example contradicts its premise — the two regions share `scratch`, so that edge is already inferred in step 1. Rather than swap in a non-overlapping pair, the step now shows the mechanism on a case the reader can still verify and says plainly that writing the edge changes nothing here. A genuinely non-overlapping example cannot be checked against a golden: the failure mode is a race, not a wrong number.

File: docs/en/user/tutorials/04-task-graph.md [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+## Step 2: an edge the inference cannot see
+
+If task B must follow task A for a reason that never shows up as a shared buffer, nothing
+derives it. Bind the producer's TaskId and name it:
+
+```python
+    with pl.at(level=pl.Level.CORE_GROUP, name_hint="stage1") as first:
+        scratch = pl.assemble(scratch, pl.add(x, x), [0, 0])
+    with pl.at(level=pl.Level.CORE_GROUP, name_hint="stage2", deps=[first]) as second:
+        out = pl.assemble(out, pl.add(scratch, scratch), [0, 0])
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/04-task-graph.md]: Fixed in 518d38c1 by the same change as the step-2 thread above — the premise and the example now agree.

File: docs/en/user/tutorials/05-scheduling-tuning.md [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+
+> **Prerequisites:** [Shaping the task graph](04-task-graph.md).
+
+## What you are building
+
+A loop you can re-run on your own kernel: **look at the graph → look at the concurrency →
+look at the resource high-water marks → change one thing → measure again.**
+
+Every step is a `RunConfig` flag that writes a file. Nothing here requires instrumenting
+your kernel.
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/tutorials/05-scheduling-tuning.md]: Fixed in 518d38c1. Now scoped to steps 1-4; steps 5 and 6 are described as reshaping the work rather than observing it.

File: docs/zh/user/tutorials/00-elementwise.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+
+要断言，不要打印。一个静默什么都没写的 kernel 产出 NaN，而 NaN 与任何值比较都不相等 —— `allclose` 能抓住它，扫一眼打印结果未必。
+
+跑完整文件：
+
+```bash
+python examples/beginner/02_elementwise.py
+```
+
+## Edge Cases
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/zh/user/tutorials/00-elementwise.md]: Fixed in 518d38c1, and swept wider than reported. All 20 zh pages using the English heading were mine across every earlier batch; the distributed chapter already used `边界情况`. All 20 now match, so the zh manual is consistent.

File: examples/advanced/03_mixed_kernel.py [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+
+    ``pl.split(pl.SplitMode.UP_DOWN)`` marks the scope as mixed: the compiler
+    halves it by rows and drives the cube and vector units concurrently.
+    """
+    with pl.at(
+        level=pl.Level.CORE_GROUP,
+        # The [128,128] FP32 tile that crosses the cube/vector boundary is 64KB.
+        # The V2C ring defaults to 8 slots of it (512KB) — far past the vector
+        # buffer limit, so the depth has to come down with it.
+        optimizations=[pl.split(pl.SplitMode.UP_DOWN), pl.cross_core_slot(slot_num=2)],
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply examples/advanced/03_mixed_kernel.py:73]: Fixed in 518d38c1. Correct — the cube produces and the vector consumes, so the ring is C2V. The V2C label came from a test whose dataflow runs the other way.

**Review by Hzfengsy — APPROVED** [review]
